### ISSUE TITLE
[cargo-clippy] Linting integration tests ignores --features

### ISSUE THREAD

**Issue by crumblingstatue** [Manishearth/rust-clippy#1266]
Cargo.toml:

``` toml
[package]
name = "whatever"
version = "0.1.0"

[features]
clippy = []

```

lib.rs:

``` Rust
#[cfg_attr(feature="clippy", allow(module_inception))]
mod foo { mod foo {} }

```

With `--features=clippy`, this compiles fine without any warnings.
Now let's add an integration test!

tests/test.rs:

``` Rust
#[test]
fn test() {}

```

Now rustc complains about unknown lint, even though we passed `--features=clippy`:

```
$ cargo clippy --features=clippy
   Compiling whatever v0.1.0 (file:///tmp/whatever)
    Finished debug [unoptimized + debuginfo] target(s) in 0.14 secs
   Compiling whatever v0.1.0 (file:///tmp/whatever)
warning: unknown lint: `module_inception`, #[warn(unknown_lints)] on by default
 --> src/lib.rs:1:36
  |
1 | #[cfg_attr(feature="clippy", allow(module_inception))]
  |                                    ^^^^^^^^^^^^^^^^

```

**Comment by bsteinb** [comment]
I do not think that `--features` is ignored, rather, when building the test, the clippy plugin is not loaded while compiling the library, but only for the test itself. Thus, with the `clippy` feature enabled, the compiler rightfully complains that it does not know about the `module_inception` lint.

**Comment by crumblingstatue** [comment]
> I do not think that --features is ignored, rather, when building the test, the clippy plugin is not loaded while compiling the library, but only for the test itself. Thus, with the clippy feature enabled, the compiler rightfully complains that it does not know about the module_inception lint.

Hmm... So this is not fixable in clippy? The only reason I'm using the `clippy` feature is to avoid flooding the output with unknown_lints warnings. Maybe I should just `allow(unknown_lints)`. I'm not sure if that is the recommended way to deal with this.

**Comment by antoyo** [comment]
I don't know if it is related, but now running `cargo clippy` gives me these warnings:
```
warning: unknown lint: `new_without_default_derive`, #[warn(unknown_lints)] on by default
warning: unknown lint: `new_without_default`, #[warn(unknown_lints)] on by default
```
I don't believe it is normal and it was not doing this before.
Thanks to fix this issue.

**Comment by mcarton** [comment]
Cc @oli-obk, is this related to  #1171?

**Comment by antoyo** [comment]
@mcarton I think this is from compiling the examples, indeed.

**Comment by oli-obk** [comment]
I think I know what's going on. For building the tests, first the crate itself is built, but without clippy enabled, because it's just a dependency. Would everyone be alright with passing `--cap-lints allow` to all dependencies while running cargo clippy?

**Comment by oli-obk** [comment]
fixed in #1332
